esp-hal/src/dma/mod.rs

```diff
@@ -948,6 +962,16 @@ where
         R::clear_in_interrupts();
         R::reset_in();
         R::set_in_descriptors(descriptors.first() as u32);
+        #[cfg(esp32s3)]
+        if peri == DmaPeripheral::Mem2Mem {
+            // Mem2Mem mode does not require a peripheral but if we leave this set to the
+            // default reset value (63) the DMA will not start and issues a
+            // descriptor error. So we set it to 0 to avoid this.
+            R::set_in_peripheral(0);
```

Comment:
Won't this potentially interfere with SPI2?

Reply:
This is what I'm not clear on.  But I was just looking at the esp32c6 TRM and it lists 10-15 as "Dummy-N" and "“Dummy-n” corresponds to register values for memory-to-memory data transfer." I'm going to test this on the s3 to see if it works.

Reply:
also since we I'm also setting 'set_mem2mem_mode()' the channel connects the tx with the rx (per TRM)

Reply:
The solution for this in esp-idf iirc, was to search all the GDMA channels to see what perpheral wasn't in use. They've fixed this hardware problem in future chips.

Reply:
Yuck, I guess I'll have to also pass in a peripheral to "steal" :-(

Reply:
Found it https://github.com/espressif/esp-idf/issues/10575.

Reply:
thanks for teh heads up!

Reply:
I've changed the implementation to accept the peripheral id.